[script.module.oauth2client] 1.4.11

robweber opened the pull request:
### Description
<!--- Provide a short summary of submitted add-on in case it's a new addition. -->
<!--- If it's plugin update only highlight biggest changes if needed. -->
<!--- Make sure you follow the checklist below before finalising your pull-request. -->

Got a pull request for this that fixes some time issues affecting other addons that use this as a dependency. Didn't see any reason this couldn't stay at the current xbmc.python version level unless you'd rather I bring it up to Jarvis or Krypton. I can re-pull to those repos if needed and adjust the value. 

### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

robweber commented:
Sorry for the 2 commits, I noticed after the fact that the Google version listed is actually 1.4.11

Razzeee commented:
While this doesn't perfectly match https://github.com/google/oauth2client/releases/tag/v1.4.11 it should be fine to merge.

But still, those are pretty old versions of the package :/

robweber commented:
Correct, there is one custom patch on top. There was an issue in the original Google implementation that caused issues parsing datetime objects with the expiry token. [Here is a link to the PR](https://github.com/robweber/script.module.oauth2client/pull/1) where I merged it in. 

My goal here was just to get things working for addons needing this as a dependency ASAP. I'm going to work on updating the whole thing with the newest from the Google repo.

Razzeee commented:
That's weird my diff to version 1.4.11 showed only two differences, none of which is the change from the PR

Razzeee commented on script.module.oauth2client/lib/oauth2client/client.py:
```diff
@@ -530,7 +532,6 @@ def authorize(self, http):
     request_orig = http.request
 
     # The closure that will replace 'httplib2.Http.request'.
-    @util.positional(1)
```
1.) This is still in 1.4.11 for me

Razzeee commented on script.module.oauth2client/lib/oauth2client/crypt.py:
```diff
@@ -38,10 +38,37 @@ class AppIdentityError(Exception):
+    ImportError if OpenSSL is unavailable.
+
+  """
+  try:
```
2.) This is how it looks in 1.4.11

````
  try:
    _ = imp.find_module('OpenSSL')
    return
  except ImportError:
    import OpenSSL
```

robweber commented:
Weird. Based on some time stamps I think I figured out what happened. Looks like the last time I updated from the source was June 19, 2015 (you're right, that is an old version!). Just looking at the crypt.py file, the tag for 1.14.11 says the last update was May 19, 2015. So well before I updated. If you look at the same file for [1.14.12](https://github.com/google/oauth2client/blob/v1.4.12/oauth2client/crypt.py) though the changes you're seeing are in that file, updated May 26, 2015. 

My guess is that I pulled this from the master branch of the oauth2client repo. The changes to the files were made post the 1.14.11 tag, but the version number in the version history wasn't updated yet. Mostly my mistake for not just pulling from the tagged branch. This addon is a dep for a different one I work on so I'm sure I was just in a hurry since I needed this to get it to function. 

How do you want me to proceed here. We can kill this whole thing and I can just update from the newest tagged if you want.

Razzeee commented:
Entirely up to you, just wanted to bring this to attention, as we prefere the modules to match upstream if possible.

MartijnKaijser commented:
preferred is from tagged version to exclude possible confusion

robweber commented:
Works for me, I'll close this myself and re-pull the most current tagged version from the source.
